src/executor/validate/mod.rs

```diff
@@ -0,0 +1,116 @@
+mod unique;
+
+use {
+    crate::{data::Row, result::Result, store::Store},
+    serde::Serialize,
+    sqlparser::ast::{ColumnDef, Ident},
+    std::{fmt::Debug, rc::Rc},
+    thiserror::Error as ThisError,
+    unique::{create_unique_constraints, fetch_all_unique_columns, specified_columns_only},
+};
+
+#[derive(ThisError, Debug, PartialEq, Serialize)]
+pub enum ValidateError {
+    #[error("conflict! storage row has no column on index {0}")]
+    ConflictOnStorageColumnIndex(usize),
+
+    #[error("duplicate entry '{0}' for unique column '{1}'")]
+    DuplicateEntryOnUniqueField(String, String),
+
+    #[error("incompatible type attempted, value: {attempted_value} used for typed column: {column_name} ({column_type})")]
+    IncompatibleTypeOnTypedField {
+        attempted_value: String,
+        column_name: String,
+        column_type: String,
+    },
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum UniqueKey {
+    Bool(bool),
+    I64(i64),
+    Str(String),
+    Null,
+}
```

Comment:
Let's make it simple :)
This is mostly related to `validate/unique.rs` so let's put inside of `validate/unique.rs`.

Then you only need to add `pub use unique::UniqueKey;` in here to export. (Because it should be accessed from `data::Value`)

Reply:
Done :-)